### PR TITLE
Java keystore support

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Issue #, if available:*

Implements the feature request in #180 

*Description of changes:*

Adds a new builder option that will get the certificate and private key from a Java keystore. Also adds a sample (including in CI) and adjusts README to show how to use the new sample that uses the Java keystore.

Requires the following CRT PR to be merged and released: https://github.com/awslabs/aws-crt-java/pull/552

__________

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java [hunk 1]
```diff
@@ -176,6 +188,58 @@ public static AwsIotMqttConnectionBuilder newMtlsWindowsCertStorePathBuilder(Str
...
+     * Create a new builder with mTLS, using a certificate and key stored in a Java keystore.
+     *
+     * @param keystorePath Path to the Java keystore on the file system.
+     * @param keystorePassword The password for the Java keystore.
+     * @param keystoreFormat The format of the Java keystore. Set to 'default' to use the default Java keystore type.
+     * @param certificateAlias The alias of the certificate and key to use with the builder.
+     * @param certificatePassword The password of the certificate and key to use with the builder.
+     * @return {@link AwsIotMqttConnectionBuilder}
+     */
+    public static AwsIotMqttConnectionBuilder newJavaKeystoreBuilder(String keystorePath, String keystorePassword, String keystoreFormat, String certificateAlias, String certificatePassword) {
```
Comment (MikeDombo) [inline sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java:201]: why not just accept the `KeyStore` directly as an input? That seems much more reasonable.

Comment (TwistedTwigleg) [reply sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java]: Hmm, good point. Changed it to use a `KeyStore` as the input and added a comment that it needs to be loaded. 👍

**Comment by TwistedTwigleg** [comment]
Just adding a quick note here: I am going to look at moving some of this Java keystore stuff to the CRT so that customers that are using the CRT can also benefit from the addition for using the Java Keystore and not just those using the SDK via the IoT Builder. Will update this PR accordingly and link to the CRT Java PR once I have it made.

**Review by xiazhvera — APPROVED** [review]

**Comment by TwistedTwigleg** [comment]
Thanks! Merging into `main`...

**Pull request merged**